File: Benchmarking/30 Euler/euler7/euler7.py
import sys

def isPrime(x):
	divisor = 2
	prime = True
	while divisor < x:
		if x % divisor == 0:
			prime = False
			break
		divisor += 1
	return prime
	
def findPrime(primeNumber):
	arrayPrimes = []
	counter = 0
	number = 2
	while counter < primeNumber:
		if isPrime(number):
			counter += 1
			if counter == primeNumber:
				break
    		number += 1
	if counter == primeNumber:
		return number

primeNumber = int(sys.argv[1])
print findPrime(primeNumber)

File: Benchmarking/30 Euler/euler7/euler7.cpp
//
//  main.cpp
//  Euler7
//
//  Created by Sonni on 19/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>
#include <vector>

std::vector<bool> getPrimes(unsigned int size)
{
    std::vector<bool> array;
    for(int i = 0; i <= size; i++)
        array.push_back(true);
    
    array[1] = false;
    for(unsigned int i = 2; i <= size; i++)
    {
        if(array[i])
        {
            unsigned int j = i + i;
            while(j <= size)
            {
                array[j] = false;
                j += i;
            }
        }
    }
    return array;
}

unsigned int findPrime(unsigned int primeNumber, unsigned int limit)
{
    std::vector<bool> primes = getPrimes(limit);
    unsigned int count = 0;
    for (unsigned int i = 1; i <= primes.size(); i++)
    {
        if (primes[i])
            count++;
        if (count == primeNumber)
            return i;
    }
    return -1;
    
}

int main(int argc, const char * argv[]) {
    std::cout << findPrime(atoi(argv[1]), atoi(argv[2])) << std::endl;;
    return 0;
}
